tests/run_compiled_diffusion_model_hotswap.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2024-present the HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This is a standalone script that checks that we can hotswap a LoRA adapter on a compiled model
+
+By itself, this script is not super interesting but when we collect the compile logs, we can check that hotswapping
+does not trigger recompilation. This is done in the TestLoraHotSwapping class in test_pipelines.py.
+
+Running this script with `check_hotswap(False)` will load the LoRA adapter without hotswapping, which will result in
+recompilation.
+
+There is an equivalent test in diffusers, see https://github.com/huggingface/diffusers/pull/9453.
+
+"""
+
+import os
+import sys
+import tempfile
+
+import torch
+from diffusers import StableDiffusionPipeline, UNet2DConditionModel
+from diffusers.utils.testing_utils import floats_tensor
+
+from peft import LoraConfig, get_peft_model_state_dict
+from peft.tuners.tuners_utils import BaseTunerLayer
+
+
+torch_device = "cuda" if torch.cuda.is_available() else "cpu"
+
+
+def get_small_unet():
```

Comment:
Could also add a note saying that currently, it does not work in the full pipeline context when compile is enabled.

Reply:
Done